storybook/figmalinkssource.cpp

```diff
@@ -38,6 +38,74 @@ FigmaLinks* FigmaLinksSource::getFigmaLinks() const
     return m_figmaLinks;
 }
 
+void FigmaLinksSource::remove(const QString &key, const QList<int> &indexes)
+{
+    if (m_filePath.isEmpty()) {
+        qWarning("FigmaLinksSource::remove - file path is not set!");
+        return;
+    }
+
+    QMap<QString, QStringList> linksMap;
+
+    if (m_figmaLinks)
+        linksMap = m_figmaLinks->getLinksMap();
+
+    auto it = linksMap.find(key);
+
+    if (it == linksMap.end()) {
+        qWarning("FigmaLinksSource::remove - provided key doesn't exist!");
+        return;
+    }
+
+    if (indexes.isEmpty())
+        return;
+
+    auto indexesSorted = indexes;
+    std::sort(indexesSorted.begin(), indexesSorted.end());
+
+    if (std::adjacent_find(indexesSorted.cbegin(), indexesSorted.cend())
+            != indexesSorted.cend()) {
+        qWarning("FigmaLinksSource::remove - provided indexes list contains duplicates!");
```

Comment:
You can use QSet to get rid of them

Reply:
It's an option. I would use QSet if it were sorted, but it's not so I ended up with solution like that. For sure it can be optimized/shortened but I don't want to do engineering here :)